src/daft-core/src/datatypes/prelude.rs

```diff
@@ -6,10 +6,15 @@ pub use super::{
 };
 
 // Import utility types and structs
-pub use super::{Field, FieldID, FieldRef, ImageFormat, ImageMode, TimeUnit};
+
+pub use daft_schema::field::{Field, FieldID, FieldRef};
+
+pub use daft_schema::image_format::ImageFormat;
+pub use daft_schema::image_mode::ImageMode;
+pub use daft_schema::time_unit::TimeUnit;
 
 // Import DataType enum
-pub use super::DataType;
+pub use daft_schema::dtype::DataType;
```

Comment:
Actually it's a good convention! Take a look at [this](https://www.lurklurk.org/effective-rust/re-export.html)

Reply:
Actually it's a good convention that we should be doing more, take a look at [this](https://www.lurklurk.org/effective-rust/re-export.html).

The TLDR is that we use a type in a public interface of this crate. we should re-export it.
This makes it very nice to actually use the library, otherwise we have to pretty much add all the transitive dependencies of the crate which defeats the purpose of the abstract.


Let's think of the case that we have the following dep tree.

```
A <- B <- C <- D
```

if we don't export types used in the public APIs the cargo toml would look like
```
D {
A, B, C
}

C {
A, B
}

B {
A
}
```